File: src/ast/PrefixExpression.java
package ast;

import java.io.StringWriter;

public class PrefixExpression extends ParentNode {
    private String _prefix;

    public PrefixExpression(String prefix, BaseNode child) {
        super(NodeType.PrefixExpression, child);
        _prefix = prefix;
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        writer.write(_prefix);
        writer.write("(");
        child.Print(writer);
        writer.write(")");
    }
}

File: src/de/fearlesstobi/demangler/ast/PrefixExpression.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class PrefixExpression(private val prefix: String, child: BaseNode?) : ParentNode(NodeType.PrefixExpression, child) {
    public override fun printLeft(writer: StringWriter) {
        writer.write(prefix)
        writer.write("(")
        child!!.print(writer)
        writer.write(")")
    }

}